net/libfab/libfab.c

```diff
@@ -3404,7 +3136,13 @@ static int libfab_buf_add(struct m0_net_buffer *nb)
 	case M0_NET_QT_MSG_SEND: {
 		M0_ASSERT(nb->nb_length <= m0_vec_count(&nb->nb_buffer.ov_vec));
 		M0_ASSERT(nb->nb_buffer.ov_vec.v_nr == 1);
-		ret = libfab_fab_ep_find(ma, NULL, nb->nb_ep->nep_addr, &ep);
+		/*
+		 * Passing name along with ptr to libfab_fab_ep_find so that
+		 * if libfab_ep_find_by_ptr() fails, we can create new endpoint
+		 * using name string.
+		 */
+		ret = libfab_fab_ep_find(ma, nb->nb_ep->nep_addr, NULL,
+					 nb->nb_ep, &ep);
```

Comment:
As we already have endpoint nb->nb_ep->nep_addr we can avoid this search in link-list. To make sure this end-point is in link-list we can add e.g. a flag in libfab element to indicate if this end-point is part of link-list, so in code where this EP is getting added and removed that flag can be updated. In this buffer add code we can check that flag.
	m0_nep_tlink_init_at_tail(net, &tm->ftm_ntm->ntm_end_points);
	m0_tl_teardown(m0_nep, &tm->ftm_ntm->ntm_end_points, net) {
		xep = libfab_ep(net);
		rc = libfab_ep_param_free(xep, tm);
	}

Reply:
Handling added in commit b85f491